enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,580 @@
+status: implementable
+see-also:
+  - enhancements/installer/connected-assisted-installer.md
+  - enhancements/installer/assisted-installer-bare-metal-validations.md
+---
+
+# Assisted Installer in-cluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Assisted Installer currently runs as a SaaS on cloud.redhat.com, enabling
+users to deploy OpenShift clusters with certain customizations, particularly on
+bare metal hardware. It is necessary to bring those capabilities on-premise in
+users' "Hub" clusters by installing clusters via Multi-cluster management, such
+as through Hive and RHACM.
+
+This enhancement proposes the first iteration of running the Assisted Installer
+in end-user clusters to meet the purpose above.
+
+## Motivation
+
+The Assisted Installer running as a SaaS has demonstrated a new and valuable
+way to install OpenShift clusters through an agent-based flow. The [Assisted
+Installer for Connected Environments](connected-assisted-installer.md)
+enhancement proposal elaborates on the value of the agent-based installations.
+
+However many customers do not want a third party service running outside of
+their network to be able to provision hosts inside their network. Reasons
+include trust, control, and reproducibility. Those users will prefer to run the
+service on-premises. There are also many customers whose network environments
+have limited or no connectivity to the internet. Finally, it is desirable to
+integrate this new installation method into RHACM, which runs on-premesis.
+
+### Goals
+
+* Expose the Assisted Installer's capabilities as a kubernetes-native API. Some
+portion may be exposed through [Hive](https://github.com/openshift/hive)'s API.
+* Enable multi-cluster management tooling to create new clusters using Assisted Installer.
+* Enable adding workers to any OpenShift cluster deployed by the Assisted Installer via
+  the same multi-cluster management tooling.
+* Enable automated creation and booting of the Assisted discovery ISO for bare-metal deployments.
+* Ensure the design is extensible for installing on other platforms.
+
+### Non-Goals
+
+* Remote Worker Node support requires solving specific problems that will be
+addressed in a separate proposal.
+* Solve central machine management. ("central machine management" involves
+running machine-api-providers on a hub cluster in order to manage Machines on
+the hub which represent Nodes in spoke clusters.) That can be done with this
+effort, or after this effort, but it is not strictly a requirement in order to
+deliver the goals stated above.
+* Run metal3 components on a non-baremetal cluster.
```

Comment:
It's mentioned elsewhere in this enhancement, but right now we are assuming that the Hub's platform is bare-metal, so that metal3 is available for booting the discovery ISOs with virtual media.  If the Hub has a different platform, this won't be possible right now.  A separate enhancement would cover how the Hub can be in a virtualized environment for example, and still manage bare metal Spoke clusters.  I agree that the way it is currently written is confusing becomes the context comes later.